CHANGES.md

```diff
@@ -1,5 +1,172 @@
-## Version 0.9.0
-
+## Version 1.0.0-MF
+
+> 2017 June 3
+
+`MF` stands for milestone final. This is the last non-RC release before 1.0.0.
+ The main purpose/focus of this release is to offer a relatively stable API to
+ work with prior to 1.0.0. It can be deemed as a proposal for the final API
+ we are going to maintain binary compatibiliy after 1.0. 
+ We will give community some time to validate it before we release 1.0.0-RC1.
+ 
+### To migrate from 0.9.0
+
+We apologize for the number of breaking changes in this release. We are trying to include
+as many breaking changes as possible in this release before we lock down the API. 
+  
+ * `cats` no longer publish the all-inclusive bundle package `"org.typelevel" % "cats"`, use `cats-core`, `cats-free`, or `cats-law` 
+   accordingly instead. If you need `cats.free`, use `"org.typelevel" % "cats-free"`, if you need `cats-laws` use 
+   `"org.typelevel" % "cats-laws"`, if neither, use `"org.typelevel" % "cats-core"`.
+ * `cats.free.Inject` is moved from `cats-free` to `cats-core` and renamed to `cats.InjectK`;
+   `cats.data.Prod` is renamed to `cats.data.Tuple2K`; `cats.data.Coproduct` is renamed to
+   `cats.data.EitherK`
+ * All `Unapply` enabled methods, e.g. `sequenceU`, `traverseU`, etc. are removed. `Unapply`
+   enabled syntax ops are also removed. Please use the partial unification SI-2712 fix
+   instead. The easiest way might be this [sbt-plugin](https://github.com/fiadliel/sbt-partial-unification).
+ *  `FunctorFilter`, `MonadCombine`, `MonadFilter`, `MonadReader`, `MonadState`, `MonadTrans`, `MonadWriter` and `TraverseFilter` are no longer in `cats`, the functionalities they provided are inhereted by the new [cats-mtl](https://github.com/edmundnoble/cats-mtl) project. Please check [here](https://github.com/edmundnoble/cats-mtl#migration-guide) for migration guide. 
+ * `CartesianBuilder` (i.e. `|@|`) syntax is deprecated, use the apply syntax on tuples instead. E.g. `(x |@| y |@| z).map(...)` should be replaced by `(x, y, z).mapN(...)`
+ * The creation methods (`left`, `right`, `apply`, `pure`, etc.) in `EitherT` were improved to take less
+   type arguments.
+ * Several `cats-core` type class instances for `cats.kernel` were moved from their companion objects to separate traits
+   and thus require imports from `cats.instances.xxx._` (or the recommended `import cats.implicits._`) now. See #1659 for more details. 
+ * `Free.suspend` is renamed to `Free.defer` for consistency. 
+ * `traverse1_`, `intercalate1` and `sequence1_` in `Reducible` were renamed to `nonEmptyTraverse_`, `nonEmptyIntercalate` and `nonEmptySequence_` respectively. 
+ * `foldLeftM` is removed from `Free`, use `foldM` on `Foldable` instead, see #1117 for detail. 
+ * `iteratorFoldM` was removed from `Foldable` due to #1716
+ * Apply syntax on tuple (e.g. `(x, y, z).map3(...)`)  was moved from `cats.syntax.tuple._` to `cats.syntax.apply._` and renamed to `mapN`, `contramapN` and `imapN` respectively.
```

Comment:
Should we move this up to after the related `CartesianBuilder` deprecation bullet?